feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameViewModel.kt

```diff
@@ -92,21 +98,30 @@ class MemorizeGameViewModel @Inject constructor(
     }
 
     override fun onItemClick(itemGameImageUiState: ItemGameImageUiState) {
-        toggleGameItem(itemGameImageUiState)
-    }
-
-    private fun toggleGameItem(itemGameImageUiState: ItemGameImageUiState) {
+        if (itemGameImageUiState.isSelected){
+            return
+        }
         viewModelScope.launch {
-            val modifyItem =
-                itemGameImageUiState.copy(isSelected = !itemGameImageUiState.isSelected)
-            val modifyBoard = (_state.value.board - itemGameImageUiState).toMutableList()
-            modifyBoard.add(modifyItem.position, modifyItem)
-            _state.update { it.copy(board = modifyBoard) }
-            delay(300)
+            toggleGameItem(itemGameImageUiState)
+            delay(500)
             saveUserPosition(itemGameImageUiState.position)
+            checkIfWinner()
         }
     }
 
+    private fun checkIfWinner() {
```

Comment:
your not checking if the user is winning or not your handling what will happen if the user won 
a function that starts with check means that the function must return a boolean and that's not what the function does
you could change the function name to "determineWinningOutcome" or something else